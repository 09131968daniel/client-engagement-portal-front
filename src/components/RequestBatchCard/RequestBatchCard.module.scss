.RequestBatchCard {
    width: 400px;
    height: 300px;
    border: 1px solid #F26925;
    box-shadow: 3px 3px 15px #72A4C260;
    margin: 30px auto;
    border-radius: 50px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 50px;
    background-color: white;
    color: #474C55;
}

.RequestBatchCardText {
    text-align: left;
    margin-bottom: 30px;
}

.RequestBatchButton {
    width: 300px;
    height: 30px;
    background-color: #F26925;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: white;
    border: 2.5px solid #F26925;
    user-select: none;
}

.RequestBatchButton:hover {
    background-color: white;
    color: #F26925;
    cursor: pointer;
}