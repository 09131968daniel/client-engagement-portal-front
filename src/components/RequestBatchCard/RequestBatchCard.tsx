import React from 'react'
import classes from './RequestBatchCard.module.scss'

interface Props {

}

const RequestBatchCard = (props: Props) => {

    const requestBatchCardText = "No batches have been selected for you just yet. Our administrators will assign you one soon.";

    return (
        <div className={classes.RequestBatchCard}>
            <div className={classes.RequestBatchCardText}>
                {requestBatchCardText}
            </div>
        </div>
    )
}

export default RequestBatchCard;
