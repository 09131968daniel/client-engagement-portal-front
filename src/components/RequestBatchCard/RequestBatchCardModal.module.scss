.RequestBatchCardModal {
    width: 350px;
    height: 40px;
    border-radius: 8px;
    background-color: #FCB414;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 0 auto;
    color: #474C55;
    border: 2px solid #F26925;
}

.XButton {
    display: flex;
    flex-direction: column;
    justify-content: center;
    position: absolute;
    margin-left: 300px;
    border: 1px solid #F26925;
    background-color: #F26925;
    border-radius: 500px;
    height: 25px;
    width: 25px;
    color: white;
    user-select: none;
}

.XButton:hover {
    cursor: pointer;
    background-color: white;
    color: #F26925;
}