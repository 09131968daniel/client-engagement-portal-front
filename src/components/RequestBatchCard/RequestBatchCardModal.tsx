import React from 'react'
import classes from './RequestBatchCardModal.module.scss'

/**
 * @function RequestBatchCardModal
 * Component to display when the client requests a new batch.
 */
const RequestBatchCardModal = () => {
    return (
        <div className={classes.RequestBatchCardModal}>
            <div>Request Sent Successfully</div>
            <div className={classes.XButton}>X</div>
        </div>
    )
}

export default RequestBatchCardModal;
