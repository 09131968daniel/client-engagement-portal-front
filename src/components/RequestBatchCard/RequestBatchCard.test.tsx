import React from 'react'
import RequestBatchCard from './RequestBatchCard'
import { shallow } from 'enzyme'

const setUp = () => {
    const component = shallow(<RequestBatchCard />);
    return component;
}

describe('RequestBatchCard Component', () => {

    let component: any;
    beforeEach(() => {
        component = setUp();
    });

    it('Should Render To The Screen', ()=>{
        expect(component.render()).toBeTruthy();
    });

    it('Should Contain Text', ()=>{
        const componentText = "No batches have been selected for you just yet. Our administrators will assign you one soon. In the meantime, youâ€™re welcome to click the button below to notify one of our administrators.";
        expect(component.text().includes(componentText)).toBe(true);
    })

    it('Should have button text', ()=>{
        expect(component.text().includes("Request Batch")).toBe(true);
    })

    it('Should Match the Snapshot', ()=>{
        expect(component).toMatchSnapshot();
    })

});
