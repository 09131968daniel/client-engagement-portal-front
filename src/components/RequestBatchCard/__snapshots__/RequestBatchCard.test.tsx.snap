// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestBatchCard Component Should Match the Snapshot 1`] = `
<div
  className="RequestBatchCard"
>
  <div
    className="RequestBatchCardText"
  >
    No batches have been selected for you just yet. Our administrators will assign you one soon.
  </div>
</div>
`;
